plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -998,10 +998,24 @@ protected boolean isSupportedJoinCondition(ConnectorSession session, JdbcJoinCon
             return false;
         }
 
-        // Remote database can be case insensitive.
-        return Stream.of(joinCondition.getLeftColumn(), joinCondition.getRightColumn())
+        boolean isVarcharJoinColumn = Stream.of(joinCondition.getLeftColumn(), joinCondition.getRightColumn())
                 .map(JdbcColumnHandle::getColumnType)
-                .noneMatch(type -> type instanceof CharType || type instanceof VarcharType);
+                .allMatch(type -> type instanceof CharType || type instanceof VarcharType);
+        if (isVarcharJoinColumn) {
+            JoinCondition.Operator operator = joinCondition.getOperator();
+            return switch (operator) {
+                case LESS_THAN, LESS_THAN_OR_EQUAL, GREATER_THAN, GREATER_THAN_OR_EQUAL -> false;
+                case EQUAL, NOT_EQUAL -> isCaseSensitiveVarchar(joinCondition.getLeftColumn()) && isCaseSensitiveVarchar(joinCondition.getRightColumn());
+                default -> false;
```

Comment:
Can this `default` be removed? If yes, switch expression will check for all values during compile time.

Reply:
it can be removed